### PR TITLE
ArgumentListWrappingRule: fix wrong indentation in `if` condition

### LINKED ISSUE #854: The linter and formatter results are inconsistent
## Expected Behavior

Trying to format the following code. Expected no errors.
```kotlin
package test

class PropertyA
class PropertyB
class PropertyC

fun someFunction(a: PropertyA, b: PropertyB, c: PropertyC): String = TODO()

fun main(args: Array<String>) {
    if (someFunction(
        PropertyA(),
        PropertyB(),
        PropertyC()
    ) ==
        "ABC"
    ) {
        TODO()
    }
}
```

## Observed Behavior

ktlintFormat task fails with the following message
```
> Task :ktlintMainSourceSetFormat FAILED
/Users/me/IdeaProjects/test/src/main/kotlin/main.kt:11:13: Unexpected indentation (expected 8, actual 12)
/Users/me/IdeaProjects/test/src/main/kotlin/main.kt:12:17: Unexpected indentation (expected 8, actual 12)
/Users/me/IdeaProjects/test/src/main/kotlin/main.kt:14:1: Unexpected indentation (expected 8, actual 12)
/Users/me/IdeaProjects/test/src/main/kotlin/main.kt:15:12: Unexpected indentation (expected 4, actual 8)
```

ktlintCheck task produces following errors:
```
> Task :ktlintMainSourceSetCheck FAILED
/Users/me/IdeaProjects/test/src/main/kotlin/main.kt:11:1: Unexpected indentation (8) (should be 12)
/Users/me/IdeaProjects/test/src/main/kotlin/main.kt:12:1: Unexpected indentation (8) (should be 12)
/Users/me/IdeaProjects/test/src/main/kotlin/main.kt:13:1: Unexpected indentation (8) (should be 12)
/Users/me/IdeaProjects/test/src/main/kotlin/main.kt:14:1: Unexpected indentation (4) (should be 8)
```

It seem to be formatted in continuation inden 8:
```kotlin
package test

class PropertyA
class PropertyB
class PropertyC

fun someFunction(a: PropertyA, b: PropertyB, c: PropertyC): String = TODO()

fun main(args: Array<String>) {
    if (someFunction(
            PropertyA(),
            PropertyB(),
            PropertyC()
        ) ==
        "ABC"
    ) {
        TODO()
    }
}
```

## Your Environment
* Version of ktlint used: 0.38.1
* Used ktlint gradleplugin (org.jlleitschuh.gradle.ktlint:9.3.0
* Version of Gradle used (if applicable): 6.3
* Operating System and version: macOS 10.15.6

### LINKED ISSUE #864: parameter-list-wrapping expects wrong indentation in returning if-else condition
## Expected Behavior
Expect this code to not cause any errors
```Kotlin
fun function() {
    return if (listOfNotNull(
            param1,
            param2,
            param3
        ).isEmpty()
    ) {
        true
    } else {
        false
    }
}
```

## Observed Behavior
```Kotlin
...
Unexpected indentation (expected 8, actual 12) (parameter-list-wrapping) // arguments
Unexpected indentation (expected 4, actual 8) (parameter-list-wrapping) // last line
```

## Steps to Reproduce
Try to run lint on the given example, in case of formatting - this gets reformatted wrongly, which then causes `indent` rule to fail

## Your Environment

* Version of ktlint used: 0.38.0
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): CLI

### PR DESCRIPTION
**Pull request by t-kameyama**
Fixes #864
Fixes #854

### CODE REVIEW

File: ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/ArgumentListWrappingRule.kt [hunk 1]
```diff
@@ -193,4 +198,10 @@ class ArgumentListWrappingRule : Rule("argument-list-wrapping") {
             .firstOrNull { it.elementType == ElementType.TYPE_PARAMETER_LIST }
             ?.children()
             ?.any { it.isWhiteSpaceWithNewline() } == true
+
+    private fun ASTNode.isOnSameLineAsIfKeyword(): Boolean {
+        val containerNode = psi.getStrictParentOfType<KtContainerNode>() ?: return false
+        val ifExpression = containerNode.parent as? KtIfExpression ?: return false
```
Comment (romtsn) [inline ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/ArgumentListWrappingRule.kt:204]: I have a feeling that `KtIfExpression` is not enough here, we should probably also consider `when` and `while`.

Comment (t-kameyama) [reply ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/ArgumentListWrappingRule.kt]: Fixed in 74656c3

File: ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/ArgumentListWrappingRule.kt [hunk 2]
```diff
@@ -206,4 +213,13 @@ class ArgumentListWrappingRule : Rule("argument-list-wrapping") {
         }
         return null
     }
+
+    private fun ASTNode.isOnSameLineAsIfKeyword(): Boolean {
+        val containerNode = psi.getStrictParentOfType<KtContainerNode>() ?: return false
+        return when (val parent = containerNode.parent) {
```
Comment (romtsn) [inline ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/ArgumentListWrappingRule.kt:219]: But what about `KtWhenExpression`?

Comment (t-kameyama) [reply ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/ArgumentListWrappingRule.kt:219]: We have a test for `when`.
https://github.com/t-kameyama/ktlint/blob/74656c35796b2541335419ba55cc9d5ba2ec54fd/ktlint-ruleset-experimental/src/test/kotlin/com/pinterest/ktlint/ruleset/experimental/ArgumentListWrappingRuleTest.kt#L366-L384

In when condition, intdent is 4.

Comment (t-kameyama) [reply ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/ArgumentListWrappingRule.kt:219]: There is a test for `when`.
https://github.com/t-kameyama/ktlint/blob/74656c35796b2541335419ba55cc9d5ba2ec54fd/ktlint-ruleset-experimental/src/test/kotlin/com/pinterest/ktlint/ruleset/experimental/ArgumentListWrappingRuleTest.kt#L366-L384

In `when` condition, intdent is 4.

Comment (romtsn) [reply ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/ArgumentListWrappingRule.kt:219]: But shouldn't it be
```Kotlin
when(foo(
        i
    )
) { ... }
```
or am I missing something?

Comment (t-kameyama) [reply ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/ArgumentListWrappingRule.kt:219]: IntelliJ formats it as follows:

```kotlin
fun foo(i: Int) = true

fun test(i: Int) {
    when (foo(
            i
    )) {
        true -> println(1)
        false -> println(2)
    }
}

```

to

```kotlin
fun foo(i: Int) = true

fun test(i: Int) {
    when (foo(
        i
    )) {
        true -> println(1)
        false -> println(2)
    }
}
```

Comment (romtsn) [reply ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/ArgumentListWrappingRule.kt:219]: gotcha, thanks for clarificaiton

**Pull request merged**